pkg/operator/provisioner/provisioner.go

```diff
@@ -141,25 +137,24 @@ func (p *RookVolumeProvisioner) Provision(options controller.VolumeOptions) (*v1
 }
 
 // createVolume creates a rook block volume.
-func createVolume(image, pool string, size int64, client rookclient.RookRestClient) (string, error) {
-	newImage := model.BlockImage{
-		Name:     image,
-		PoolName: pool,
-		Size:     uint64(size),
+func (p *RookVolumeProvisioner) createVolume(image, pool string, size int64) error {
+	if image == "" || pool == "" || size == 0 {
+		return fmt.Errorf("image missing required fields (image=%s, pool=%s, size=%d)", image, pool, size)
 	}
 
-	res, err := client.CreateBlockImage(newImage)
+	createdImage, err := ceph.CreateImage(p.context, p.provConfig.clusterName, image, pool, uint64(size))
 	if err != nil {
-		return "", fmt.Errorf("Failed to create rook block image %s/%s: %v", pool, image, err)
+		return fmt.Errorf("Failed to create rook block image %s/%s: %v", pool, image, err)
 	}
+	logger.Infof("Rook block image created: %s", createdImage.Name)
 
-	return res, nil
+	return nil
 }
 
 // Delete removes the storage asset that was created by Provision represented
 // by the given PV.
 func (p *RookVolumeProvisioner) Delete(volume *v1.PersistentVolume) error {
-	rookClient, err := api.GetRookClient(p.provConfig.clusterNamespace, p.client)
+	rookClient, err := api.GetRookClient(p.provConfig.clusterNamespace, p.context.Clientset)
```

Comment:
The Delete() is also using the client. Can you remove the dependency here too or were you planning on doing this in a different PR?